modules/polygon/src/polygon-utils.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable max-statements, max-depth, complexity, no-unused-expressions */
+
+import {equals} from '@math.gl/core';
+
+/** @type {typeof import('./polygon-utils').WINDING} */
+export const WINDING = {
+  CLOCKWISE: 1,
+  COUNTER_CLOCKWISE: -1
+};
+
+/** @type {typeof import('./polygon-utils').modifyPolygonWindingDirection} */
+export function modifyPolygonWindingDirection(points, direction) {
+  const currentDirection = getPolygonWindingDirection(points);
+  if (currentDirection !== direction) {
+    points.reverse();
+  }
+}
+
+/** @type {typeof import('./polygon-utils').getPolygonWindingDirection} */
+export function getPolygonWindingDirection(points) {
+  return Math.sign(getPolygonSignedArea(points));
+}
+
+/** @type {typeof import('./polygon-utils').getPolygonSignedArea} */
+export function getPolygonSignedArea(points) {
+  // https://en.wikipedia.org/wiki/Shoelace_formula
+  let area = 0;
+  forEachSegmentInPolygon(points, (p1, p2) => {
+    area += areaCalcCallback(p1[0], p1[1], p2[0], p2[1]);
+  });
+  return area / 2;
+}
+
+/** @type {typeof import('./polygon-utils').forEachSegmentInPolygon} */
+export function forEachSegmentInPolygon(points, visitor) {
+  const numPoints = points.length;
+  for (let i = 0; i < numPoints - 1; ++i) {
+    visitor(points[i], points[i + 1], i, i + 1);
+  }
+
+  const isClosed = equals(points[points.length - 1], points[0]);
+  if (!isClosed) {
+    visitor(points[numPoints - 1], points[0], numPoints - 1, 0);
+  }
+}
+
+/** @type {typeof import('./polygon-utils').modifyPolygonWindingDirectionFlat} */
+export function modifyPolygonWindingDirectionFlat(points, start, end, size, direction) {
+  const windingDirection = getPolygonWindingDirectionFlat(points, start, end, size);
+  if (windingDirection !== direction) {
+    reversePolygonFlat(points, start, end, size);
+  }
+}
+
+/** @type {typeof import('./polygon-utils').getPolygonWindingDirectionFlat} */
+export function getPolygonWindingDirectionFlat(points, start, end, size) {
+  return Math.sign(getPolygonSignedAreaFlat(points, start, end, size));
+}
+
+/** @type {typeof import('./polygon-utils').getPolygonSignedAreaFlat} */
+export function getPolygonSignedAreaFlat(points, start, end, size) {
+  // https://en.wikipedia.org/wiki/Shoelace_formula
+  let area = 0;
+  forEachSegmentInPolygonFlat(points, start, end, size, (p1x, p1y, p2x, p2y) => {
+    area += areaCalcCallback(p1x, p1y, p2x, p2y);
+  });
+  return area / 2;
+}
+
+/** @type {typeof import('./polygon-utils').forEachSegmentInPolygonFlat} */
+export function forEachSegmentInPolygonFlat(points, start, end, size, visitor) {
+  const numPoints = (end - start) / size;
+  for (let i = 0; i < numPoints - 1; ++i) {
+    visitor(
+      points[start + i * size],
+      points[start + i * size + 1],
+      points[start + (i + 1) * size],
+      points[start + (i + 1) * size + 1],
+      i,
+      i + 1
+    );
+  }
+
+  const endPointIndex = start + (numPoints - 1) * size;
```

Comment:
It would be good if this function offers a parameter to skip the isClosed check. In deck.gl all polygons are already closed before tessellation.